sw/device/tests/pwrmgr_deep_sleep_all_wake_ups.c

```diff
@@ -53,48 +37,34 @@ bool test_main(void) {
   ret_sram_testutils_init();
 
   init_units();
-  // Enable all AST alerts in sensor_ctrl
-  for (uint32_t k = 0; k < SENSOR_CTRL_PARAM_NUM_ALERT_EVENTS; k++) {
-    CHECK_DIF_OK(
-        dif_sensor_ctrl_set_alert_en(&sensor_ctrl, k, kDifToggleEnabled));
-  }
-
-  // Enable all the AON interrupts used in this test.
-  rv_plic_testutils_irq_range_enable(&rv_plic, kTopEarlgreyPlicTargetIbex0,
-                                     kTopEarlgreyPlicIrqIdPwrmgrAonWakeup,
-                                     kTopEarlgreyPlicIrqIdPwrmgrAonWakeup);
-
-  // Enable pwrmgr interrupt.
-  CHECK_DIF_OK(dif_pwrmgr_irq_set_enabled(&pwrmgr, 0, kDifToggleEnabled));
 
   uint32_t wakeup_unit = 0;
 
-  if (UNWRAP(pwrmgr_testutils_is_wakeup_reason(&pwrmgr, 0)) == true) {
+  if (UNWRAP(pwrmgr_testutils_is_wakeup_reason(&pwrmgr, 0))) {
     LOG_INFO("POR reset");
     CHECK_STATUS_OK(ret_sram_testutils_counter_clear(kCounterCases));
-    CHECK_STATUS_OK(
-        ret_sram_testutils_counter_get(kCounterCases, &wakeup_unit));
-    execute_test(wakeup_unit, /*deep_sleep=*/true);
   } else {
     CHECK_STATUS_OK(
         ret_sram_testutils_counter_get(kCounterCases, &wakeup_unit));
     check_wakeup_reason(wakeup_unit);
     LOG_INFO("Woke up by source %d", wakeup_unit);
     clear_wakeup(wakeup_unit);
+    delay_n_clear(4);
     CHECK_STATUS_OK(ret_sram_testutils_counter_increment(kCounterCases));
+  }
+
+  while (true) {
     CHECK_STATUS_OK(
         ret_sram_testutils_counter_get(kCounterCases, &wakeup_unit));
-    if (wakeup_unit >= PWRMGR_PARAM_NUM_WKUPS) {
+    if (wakeup_unit >= get_wakeup_count()) {
       return true;
-    } else if (kDeviceType != kDeviceSimDV &&
-               wakeup_unit == PWRMGR_PARAM_ADC_CTRL_AON_WKUP_REQ_IDX) {
-      // Skip ADC_CTRL and sensor control if not in sim_dv.
+    }
+    if (execute_test(wakeup_unit, /*deep_sleep*/ true)) {
```

Comment:
Up to you, but this is a pervasive idiom.
```suggestion
    if (execute_test(wakeup_unit, /*deep_sleep=*/true)) {
```